internal/resource/url.go

```diff
@@ -542,6 +543,8 @@ func (f *Fetcher) uncompress(r io.Reader, opts FetchOptions) (io.ReadCloser, err
 		return ioutil.NopCloser(r), nil
 	case "gzip":
 		return gzip.NewReader(r)
+	case "base64":
```

Comment:
I'd prefer that we not do this here, and handle base64-decoding in the Azure provider instead.  Otherwise this function will be a mix of cases that are supported by the `Resource.Compression` spec field and cases that are supported only for internal convenience.